src/pages/admin/donate/components/donate-page-content/DonatePageContent.tsx

```diff
@@ -0,0 +1,291 @@
+
+        setIsSupportOptionsLoading(true);
+        SupportOptionsApi.getAll(client, bankCurrency)
+            .then((data) => {
+                if (isAlive) {
+                    setSupportOptions(data);
+                }
+            })
+            .catch(() => {
+                if (isAlive) {
+                    setSupportOptions([]);
+                }
+            })
+            .finally(() => {
+                if (isAlive) {
+                    setIsSupportOptionsLoading(false);
+                }
+            });
+
+        return () => {
+            isAlive = false;
+        };
+    }, [client, selectedCategory]);
+
+    const handleCategorySelect = useCallback(
+        (currency: Currencies) => {
+            if (currency === selectedCategory) {
+                return;
+            }
+
+            setSupportOptions([]);
+            setIsSupportOptionsLoading(true);
+            setSelectedCategory(currency);
+        },
+        [selectedCategory],
+    );
+
+    // Support Options handlers
+    const handleCreateSupportOption = useCallback(
+        async (name: string, value: string) => {
+            const bankCurrency = mapCurrencyToBankCurrency(selectedCategory);
+            setIsSupportOptionsLoading(true);
+            try {
+                const newOption = await SupportOptionsApi.create(client, { name, value, currency: bankCurrency });
+                setSupportOptions((prev) => [...prev, newOption]);
+            } finally {
+                setIsSupportOptionsLoading(false);
+            }
+        },
+        [client, selectedCategory],
+    );
+
+    const handleUpdateSupportOption = useCallback(
+        async (id: number, name: string, value: string) => {
+            setIsSupportOptionsLoading(true);
+            try {
+                const updatedOption = await SupportOptionsApi.update(client, id, { name, value });
+                setSupportOptions((prev) => prev.map((option) => (option.id === id ? updatedOption : option)));
+            } finally {
+                setIsSupportOptionsLoading(false);
+            }
+        },
+        [client],
+    );
+
+    const handleDeleteSupportOption = useCallback(
+        async (id: number) => {
+            setIsSupportOptionsLoading(true);
+            try {
+                await SupportOptionsApi.delete(client, id);
+                setSupportOptions((prev) => prev.filter((option) => option.id !== id));
+            } finally {
+                setIsSupportOptionsLoading(false);
+            }
+        },
+        [client],
+    );
+
+    // Bank Details handlers
+    const handleCreateBankDetails = useCallback(
+        async (data: any) => {
+            if (!config) return;
+            try {
+                const { correspondentBanks: _, ...bankData } = data;
+                const newItem = await config.create(client, bankData);
+                setItems((prev: any) => [...prev, { ...newItem, correspondentBanks: [] }]);
```

Comment:
I guess here shouldn't be `any`, please check
```suggestion
                setItems((prev) => [...prev, { ...newItem, correspondentBanks: [] }]);
```